### PR TITLE
Issues/reader multitasking

### PR DESCRIPTION
**Pull request by aerych**
This PR includes a few layout tweaks to improve support for iOS 9 Multitasking in the Reader list, detail and comments, and also in the my sites Post list. 

Repeat this test with the Reader List, Reader Detail, Reader Comments, and My Sites Post List

To test:
1. Launch the app on an iPad Air 2 (Simulator)
2. Open the feature your testing
3. Try rotating the device
4. Switch to Safari
5. Try "Slide Over". Attempt to rotate the device
6. Resize WPiOS's screen, so that it covers 50% of the screen
7. Try rotating again

Note: DTCoreText is notoriously glitchy in the comment cells running in the Simulator.  If you see any bizarre artifacting its a known issue and (so far) has not reproduced on a device. (Can't wait to kiwf!) 

@jleandroperez Since you're already fussed with multitasking for Notifications would you mind reviewing this one? 

Needs review: @jleandroperez

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 1]
```diff
@@ -203,7 +183,9 @@ - (void)didRotateFromInterfaceOrientation:(UIInterfaceOrientation)fromInterfaceO
     [super didRotateFromInterfaceOrientation:fromInterfaceOrientation];
 
     NSIndexPath *selectedIndexPath = [self.tableView indexPathForSelectedRow];
-    if (IS_IPHONE) {
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: Bye bye, IS_IPHONE!

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 2]
```diff
@@ -136,13 +137,48 @@ import Foundation
...
+        if needsRefreshCachedCellHeightsBeforeLayout {
+            needsRefreshCachedCellHeightsBeforeLayout = false
+
+            let width = view.frame.width
+            tableViewHandler.refreshCachedRowHeightsForWidth(width)
+            tableView.reloadRowsAtIndexPaths(tableView.indexPathsForVisibleRows!, withRowAnimation: .None)
+        }
+    }
+
+    @available(iOS 8.0, *)
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: Wooow this one is new to me. Is that what i think it is? an annotation marking the method "visible only" to iOS 8?

**Comment by jleandroperez** [comment]
Tested Side Over, and Side by Side (iPad Air 2 Simulator).

Looking / reacting great to Window dimension changes!.

:shipit:

**Comment by aerych** [comment]
Woot!  Thanks @jleandroperez!

**Pull request merged**